extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/AdditionalRouteDescription.java

```diff
@@ -31,6 +31,23 @@ public void setUri(String uri) {
 
     @Override
     public int compareTo(AdditionalRouteDescription o) {
-        return uri.compareTo(o.uri);
+        int slashCount = 0;
```

Comment:
Rather than adding a compareTo on this object, we can just sort the AdditionalRouteDescriptions where we use them with something like:

```
List<AdditionalRouteDescription> routeDescriptions; 
// ...
        routeDescriptions.sort(Comparator
                .comparingInt((AdditionalRouteDescription desc) -> desc.getUri().split("/").length)
                .thenComparing(Comparator.comparing(AdditionalRouteDescription::getUri)));
```

I have not tested the above

Reply:
Thanks. I'll do so.

Reply:
Yes I think we can add it to all media types.